lib/SPIRVProducerPass.cpp

```diff
@@ -83,6 +83,9 @@ cl::opt<bool>
     ShowProducerIR("show-producer-ir", cl::init(false), cl::ReallyHidden,
                    cl::desc("Dump the IR at the start of SPIRVProducer"));
 
+cl::opt<std::string> SPIRVVersionOpt("spv-version", cl::init("1.0"), cl::Hidden,
```

Comment:
I've been thinking about this some more and it would be better to use an enum here I think.

```
enum SPIRVVersion {
  SPIRV_1_0,
  SPIRV_1_3,
  SPIRV_1_5, // note that lack of command line option to specify it
}

cl::option<SPIRVVersion> version("spv-version", cl::init(SPIRV_1_0),
                                  cl::values(
                                    clEnumValN(SPIRV_1_0, "1.0", "SPIR-V version 1.0 (Vulkan 1.0)")
                                    clEnumValN(SPIRV_1_3, "1.3", "SPIR-V version 1.3 (Vulkan 1.1). Experimental")),
                                  cl::desc("Specify the SPIR-V binary version"));              
```

LLVM will do all the parsing and generate a reasonable error. Then just have a method that converts to the packed uint32_t version for the binary and use the enum everywhere else.

Also, I'm sort of leaning to towards exposing the option in the help and the docs now and not supporting inference. We can defer the frontend error checking for now. Sorry for the change on this.

Reply:
Sounds good. Also like it better to use the supported version from the underlying vulkan driver. Will need some changes in clvk.